processing/src/main/java/org/apache/carbondata/processing/datatypes/PrimitiveDataType.java

```diff
@@ -344,7 +344,7 @@ public int getSurrogateIndex() {
             byte[] value = null;
             if (isDirectDictionary) {
               int surrogateKey;
-              if (!(input instanceof Long)) {
+              if (!(input instanceof Long) && !(input instanceof Integer)) {
```

Comment:
Does other datatype have the same problem?  such as short, byte

Reply:
no, it is problem with date only